src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -121,7 +122,17 @@ protected void ShutdownAllNodes(long hostHandshake, long clientHandshake, NodeCo
             // For all processes in the list, send signal to terminate if able to connect
             foreach (Process nodeProcess in nodeProcesses)
             {
-                Stream nodeStream = TryConnectToProcess(nodeProcess.Id, 30/*verified to miss nodes if smaller*/, hostHandshake, clientHandshake);
+                // Verified to miss nodes if smaller.
+                int timeout = 30;
```

Comment:
It might be better to start nodes with the same priority as normal but quickly switch to low priority afterwards.
1) Nodes started with low priority are more likely to be killed between when they're started and when they finish the handshake. This leads to the hang in [this issue](https://github.com/microsoft/msbuild/issues/860)
2) If there's a 30-second timeout, and the node gets swapped out before then, this will fail.
3) Node creation is expected to be fast (<< 30 seconds), so a brief moment of high priority will likely be unnoticed.

I started a change to make node creation high priority, but I haven't finished it yet. It might be reasonable to combine the two if you like the idea of starting at high priority.